.github/workflows/merge-to-main.yml

```
@@ -0,0 +1,85 @@
+          git fetch origin main dev
+          AHEAD=$(git rev-list --count origin/main..origin/dev)
+          echo "commits_ahead=$AHEAD" >> $GITHUB_OUTPUT
+          if [ "$AHEAD" -eq 0 ]; then
+            echo "❌ Dev branch is not ahead of main. Nothing to merge."
+            exit 1
+          else
+            echo "✅ Dev is $AHEAD commits ahead of main"
+          fi
+
+      - name: Check for merge conflicts
+        run: |
```

actionlint (1.7.7)

shellcheck reported issue in this script: SC2046:warning:3:16: Quote this to prevent word splitting

(shellcheck)